Core/src/Geometry/GenericApproachDescriptor.cpp

```diff
@@ -23,25 +23,25 @@ void Acts::GenericApproachDescriptor::registerLayer(const Layer& lay) {
   }
 }
 
-Acts::ObjectIntersection<Acts::Surface>
-Acts::GenericApproachDescriptor::approachSurface(
+Acts::SurfaceIntersection Acts::GenericApproachDescriptor::approachSurface(
     const GeometryContext& gctx, const Vector3& position,
     const Vector3& direction, const BoundaryCheck& bcheck) const {
   // almost always 2
-  boost::container::small_vector<ObjectIntersection<Surface>, 2> sIntersections;
+  boost::container::small_vector<SurfaceIntersection, 2> sIntersections;
   sIntersections.reserve(m_surfaceCache.size());
-  for (auto& sf : m_surfaceCache) {
+  for (auto sf : m_surfaceCache) {
```

Comment:
Surface copy on purpose?
```suggestion
  for (const auto& sf : m_surfaceCache) {
```
?